### PR TITLE
Document RD when air-gapped.

### PR DESCRIPTION
**Pull request by ericpromislow**
This is a first draft. I can continue to work on it, or happy to hand it over to anyone on the docs team.

### CODE REVIEW

File: docs/how-to-guides/running-air-gapped.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+or for Windows:
+
+`https://dl.k8s.io/VERSION/bin/PLATFORM/CPU/kubectl.exe`
+
+`VERSION` will have the form `vMAJOR.MINOR.PATCH` (like `v1.22.1`),
+
+`PLATFORM` will be one of `darwin`, `linux`, or `windows`,
+
+`CPU` will `arm64` on M1 machines and `amd64` everywhere else.
```
Comment (btat) [inline docs/how-to-guides/running-air-gapped.md:127]:
```suggestion
The location of this directory is more straightforward. On all platforms, it's at `HOME/.kuberlr/PLATFORM-ARCH`, where:

 - `HOME` is the home directory is usually `%HOMEDRIVE%\%HOMEPATH` on Windows, and `~` or `$HOME` on macOS and Linux.
 - `PLATFORM` is one of `windows`, `linux`, or `darwin`.
 - `ARCH` is `aarch64` on M1 machines, and `amd64` everywhere else.

To populate it, determine which versions of Kubernetes you'll be using, and download the appropriate executables from the internet. These would be in:

<Tabs groupId="os">
  <TabItem value="Windows">
  
`https://dl.k8s.io/VERSION/bin/PLATFORM/CPU/kubectl.exe`

  </TabItem>
  <TabItem value="macOS & Linux"
  
`https://dl.k8s.io/VERSION/bin/PLATFORM/CPU/kubectl`

  </TabItem>
</Tabs>

Where:

- `VERSION` will have the form `vMAJOR.MINOR.PATCH` (like `v1.22.1`),
- `PLATFORM` will be one of `darwin`, `linux`, or `windows`,
- `CPU` will `arm64` on M1 machines and `amd64` everywhere else.
```

**Review by btat — CHANGES_REQUESTED** [review]
Added some formatting suggestions. 

I suggested tabs for the instructions/commands that vary slightly based on OS. That way only the relevant info is presented and there's less scrolling for the user. (Tab choices are also remember across pages)

An entry needed to be added to the sidebars.js file in order for it to show up on the navigation bar. I commit this bit directly.

File: docs/how-to-guides/running-air-gapped.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
+---
+title: Running When Offline
+---
+
+Rancher Desktop can be run when offline, aka in air-gapped mode. This document covers requirements
+and possible problems when running in air-gapped mode.
+
+### Network-sensitive areas
```
[the remark this answers is not in the document]
Comment (ericpromislow) [reply docs/how-to-guides/running-air-gapped.md:8]: done

File: docs/how-to-guides/running-air-gapped.md [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+You _could_ install versioned `kubectl`s manually. This is covered in the next section.
+
+### Preparing an Air-Gapped System
+
+Here we assume you have some kind of removable media which you can populate on an internet-connected system, and then move over to the air-gapped system.
+
+There are two directories that need to be populated in order for Rancher Desktop to function off-line:
+
+#### The Cache directory
```
[the remark this answers is not in the document]
Comment (ericpromislow) [reply docs/how-to-guides/running-air-gapped.md:59]: done

File: docs/how-to-guides/running-air-gapped.md [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+On Linux, the cache directory is at `$HOME/.cache/rancher-desktop` and the commands to populate it would be
+
+```
+CACHEDIR=$HOME/.cache/rancher-desktop
+mkdir -p $CACHEDIR/k3s
+cp $SOURCEDISK/k3s-versions.json $CACHEDIR/
+cp -r $SOURCEDISK/v<MAJOR>.<MINOR>.<PATCH>+k3s<BUILD> $CACHEDIR/k3s/
+```
+
+#### The kuberlr directory
```
[the remark this answers is not in the document]
Comment (ericpromislow) [reply docs/how-to-guides/running-air-gapped.md:111]: done

File: docs/how-to-guides/running-air-gapped.md [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+
+`CPU` will `arm64` on M1 machines and `amd64` everywhere else.
+
+For example, to get a kubectl for Windows that will work with Kubernetes 1.22, this Windows command-shell (not powershell) command will put it on the `SOURCEDISK`:
+
+```
+wget -O %SOURCEDISK/kubectl1.22.1.exe https://dl.k8s.io/v1.22.1/bin/windows/amd64/kubectl.exe
+```
+
+##### A note on versions
```
[the remark this answers is not in the document]
Comment (ericpromislow) [reply docs/how-to-guides/running-air-gapped.md:135]: done

File: docs/how-to-guides/running-air-gapped.md [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+
+There are two directories that need to be populated in order for Rancher Desktop to function off-line:
+
+#### The Cache directory
+
+To populate a source disk (which we refer to here as %SOURCEDISK%, although it is probably some kind of removable medium like a USB thumb drive), you need the following files:
+
+* `k3s-versions.json` -- this file is created by Rancher Desktop. It reads a raw JSON file from `https://update.k3s.io/v1-release/channels` and converts it into a different kind of JSON file. Currently there is no utility to do that conversion; the easiest way to get this file is to run Rancher Desktop on a connected system and save the `CACHE/k3s-versions.json` file (see below for where `CACHE` exists on different platforms).
+* Tar archives of Kubernetes k3s images. These are listed at https://github.com/k3s-io/k3s/releases, and you'll want to download `k3s-airgap-images-amd64.tar` or `k3s-airgap-images-arm64.tar` (for AMD/intel and M1 machines respectively) for the versions you plan on working with. For example, the following commands will let you work with K3s v 1.24.3 build 1 offline:
```
[the remark this answers is not in the document]
Comment (ericpromislow) [reply docs/how-to-guides/running-air-gapped.md:64]: done

File: docs/how-to-guides/running-air-gapped.md [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+
+### Preparing an Air-Gapped System
+
+Here we assume you have some kind of removable media which you can populate on an internet-connected system, and then move over to the air-gapped system.
+
+There are two directories that need to be populated in order for Rancher Desktop to function off-line:
+
+#### The Cache directory
+
+To populate a source disk (which we refer to here as %SOURCEDISK%, although it is probably some kind of removable medium like a USB thumb drive), you need the following files:
```
[the remark this answers is not in the document]
Comment (ericpromislow) [reply docs/how-to-guides/running-air-gapped.md:61]: done

File: docs/how-to-guides/running-air-gapped.md [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+
+`https://dl.k8s.io/VERSION/bin/PLATFORM/CPU/kubectl.exe`
+
+`VERSION` will have the form `vMAJOR.MINOR.PATCH` (like `v1.22.1`),
+
+`PLATFORM` will be one of `darwin`, `linux`, or `windows`,
+
+`CPU` will `arm64` on M1 machines and `amd64` everywhere else.
+
+For example, to get a kubectl for Windows that will work with Kubernetes 1.22, this Windows command-shell (not powershell) command will put it on the `SOURCEDISK`:
```
[the remark this answers is not in the document]
Comment (ericpromislow) [reply docs/how-to-guides/running-air-gapped.md:129]: done

File: docs/how-to-guides/running-air-gapped.md [hunk 9]
```diff
@@ -0,0 +1,137 @@
+---
+title: Running When Offline
+---
+
```
[the remark this answers is not in the document]
Comment (ericpromislow) [reply docs/how-to-guides/running-air-gapped.md:4]: Done. Guess this is some kind of magic markup that our markdown processor knows to process.

File: docs/how-to-guides/running-air-gapped.md [hunk 10]
```diff
@@ -0,0 +1,137 @@
...
+```
+cd .../CACHE
+mkdir v1.24.3+k3s1
+cd v1.24.3+k3s1
+wget https://github.com/k3s-io/k3s/releases/download/v1.24.3%2Bk3s1/k3s-airgap-images-amd64.tar
+wget https://github.com/k3s-io/k3s/releases/download/v1.24.3%2Bk3s1/sha256sum-amd64.txt
+```
+
+##### Windows
```
[the remark this answers is not in the document]
Comment (ericpromislow) [reply docs/how-to-guides/running-air-gapped.md:74]: Done

File: docs/how-to-guides/running-air-gapped.md [hunk 11]
```diff
@@ -0,0 +1,137 @@
...
+```
+
+Assuming you have some source media, you would also run the following commands to pre-populate the cache:
+
+```
+copy-item %SOURCEDISK%\k3s-versions.json %HOME%\AppData\Local\rancher-desktop\cache\
+copy-item -Recurse %SOURCEDISK%\v<MAJOR>.<MINOR>.<PATCH>+k3s<BUILD> %HOME%\AppData\Local\rancher-desktop\cache\k3s\
+```
+
+##### macOS
```
[the remark this answers is not in the document]
Comment (ericpromislow) [reply docs/how-to-guides/running-air-gapped.md:89]: done

File: docs/how-to-guides/running-air-gapped.md [hunk 12]
```diff
@@ -0,0 +1,137 @@
...
+On macOS, the cache directory is at `$HOME/Library/Caches/rancher-desktop` and the commands to populate it would be
+
+```
+CACHEDIR=$HOME/Library/Caches/rancher-desktop
+mkdir -p $CACHEDIR/k3s
+cp $SOURCEDISK/k3s-versions.json $CACHEDIR/
+cp -r $SOURCEDISK/v<MAJOR>.<MINOR>.<PATCH>+k3s<BUILD> $CACHEDIR/k3s/
+```
+
+##### Linux
```
[the remark this answers is not in the document]
Comment (ericpromislow) [reply docs/how-to-guides/running-air-gapped.md:100]: done

File: docs/how-to-guides/running-air-gapped.md [hunk 13]
```diff
@@ -0,0 +1,137 @@
...
+
+On Linux, the cache directory is at `$HOME/.cache/rancher-desktop` and the commands to populate it would be
+
+```
+CACHEDIR=$HOME/.cache/rancher-desktop
+mkdir -p $CACHEDIR/k3s
+cp $SOURCEDISK/k3s-versions.json $CACHEDIR/
+cp -r $SOURCEDISK/v<MAJOR>.<MINOR>.<PATCH>+k3s<BUILD> $CACHEDIR/k3s/
+```
+
```
[the remark this answers is not in the document]
Comment (ericpromislow) [reply docs/how-to-guides/running-air-gapped.md:110]: done

File: docs/how-to-guides/running-air-gapped.md [hunk 14]
```diff
@@ -0,0 +1,137 @@
...
+In this case, we're discussing a system that was initialized when connected to the internet, but we're about
+to take the machine offline for future use.
+
+Suppose there are three versions of `k3s` in the `rancher-desktop` cache.
+
+1.24.3
+
+1.21.14
+
+1.19.16
```
[the remark this answers is not in the document]
Comment (ericpromislow) [reply docs/how-to-guides/running-air-gapped.md:34]: done

File: docs/how-to-guides/running-air-gapped.md [hunk 15]
```diff
@@ -0,0 +1,137 @@
...
+
+But suppose that on this system we only ran `kubectl` when using versions `1.24.3` and `1.21.14`. This means that 
+the `~/.kuberlr/PLATFORM-ARCH/` directory (`$HOME/windows-amd64` on Windows) will contain only two files:
+
+kubectl1.24.3
+
+kubectl1.21.14
```
[the remark this answers is not in the document]
Comment (ericpromislow) [reply docs/how-to-guides/running-air-gapped.md:41]: done

File: docs/how-to-guides/running-air-gapped.md [hunk 16]
```diff
@@ -0,0 +1,137 @@
...
+Rancher Desktop can be run when offline, aka in air-gapped mode. This document covers requirements
+and possible problems when running in air-gapped mode.
+
+### Network-sensitive areas
+
+There are two areas where Rancher Desktop assumes network availability and will recover in an air-gapped situation:
+
+1. Pulling Kubernetes `k3s` images into the `k3s` cache directory
+
+2. Using `kuberlr` as a version-aware wrapper around `kubectl`, so the client never differs from the Kubernetes serer by more than one minor version.
```
[the remark this answers is not in the document]
Comment (ericpromislow) [reply docs/how-to-guides/running-air-gapped.md:14]: Good catch - my IDE didn't flag that, and I missed it in proofreading

File: docs/how-to-guides/running-air-gapped.md (hunk 1 above)
Comment (ericpromislow) [reply docs/how-to-guides/running-air-gapped.md:127]: Better, but I made a slight change to the wording in the line describing `HOME`

**Review by btat — APPROVED** [review]

**Pull request merged**